pkg/controller/metadata/metadata_controller.go

```diff
@@ -347,6 +349,40 @@ func (r *ReconcileMetadata) generateMetadataSecretData(instance *verifyv1beta1.M
 	return data, nil
 }
 
+// This function determins if we should regenerate the metadata or not.
+func ShouldRegenerate(secretsObj *corev1.Secret, hashOfRequestSpec string) bool {
+	secretsMap := secretsObj.Data
+
+	// If the rest of the config has changed and the hash has changed, regenerate regardless.
+	if secretsObj.ObjectMeta.Annotations[VersionAnnotation] != hashOfRequestSpec {
+		return true
+	}
+
+	// Checking the date from the secrets store to see if we need to regenerate the metadata.
+	byteValidityDays := secretsMap[ValidityDays]
+	byteValidUntil := secretsMap[ValidUntil]
+
+	// If theres nothing in the map the regenerate metadata.
+	if byteValidityDays == nil || byteValidUntil == nil {
```

Comment:
Not sure about this but I guess this is a reasonable thing to do

Reply:
😱 this is the alternative https://stackoverflow.com/a/33618772